src/Components/Router.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "Component.h"
+#include <vector>
+#include <Winsock2.h>
+
+class Controller;
+class Decoder;
+
+class Router: public Component<H264> {
+public:
+    Router(Decoder*, Controller*);
+    bool setPort(int);
+    bool link(string, int, int);
+protected:
+    void start();
+    bool wait();
+    void process();
+    void end();
+private:
+    Decoder* decoder;
+    Controller* controller;
+    atomic_bool living;
+    SOCKET upStream, entry;
+    vector<SOCKET> downStream;
+    mutex vecMtx;
```

Comment:
Please name the variables according to the style guide:
https://google.github.io/styleguide/cppguide.html#Variable_Names
(vec_mtx vs vecMtx)